content/learning-paths/servers-and-cloud-computing/onnx-on-azure/deploy.md

```diff
@@ -0,0 +1,135 @@
+distribution created by Microsoft. 
+To know more about Azure Linux 3.0, kindly refer [What is Azure Linux Container Host for AKS](https://learn.microsoft.com/en-us/azure/azure-linux/intro-azure-linux). Azure Linux 3.0 offers support for Aarch64. However, the standalone VM image for Azure Linux 3.0 or CBL Mariner 3.0 is not available for Arm.
+
+Hence, to use the default software stack provided by the Microsoft team, this guide will focus on creating a docker container with Azure Linux 3.0 as a base image and will build 
+and run the onnx based application inside the container. 
+
+### Create Azure Linux 3.0 Docker Container 
+The [Microsoft Artifact Registry](https://mcr.microsoft.com/en-us/artifact/mar/azurelinux/base/core/about) offers updated docker image for the Azure Linux 3.0.  
+
+To create a docker container, install docker, and then follow the below instructions: 
+
+```console
+$ sudo docker run -it --rm mcr.microsoft.com/azurelinux/base/core:3.0
+``` 
+
+The default container startup command is bash. tdnf and dnf are the default package managers.
+
+### Install Python and Virtual Environment:
+
+```console
+$ tdnf update
+$ tdnf install -y python3 python3-pip python3-virtualenv
+```
+Create and activate a virtual environment:
+
+```console
+$ python3 -m venv onnx-env
+$ source onnx-env/bin/activate
+```
+Note: Using a virtual environment isolates ONNX and its dependencies to avoid system conflicts.
+
+### Install ONNX and Required Libraries:
+
+```console
+$ pip install --upgrade pip
+$ pip install onnx onnxruntime fastapi uvicorn numpy
+```
+This installs ONNX libraries along with FastAPI (web serving) and NumPy (for input tensor generation).
+
+Validate ONNX and ONNX Runtime:
+Create **version.py** with below codes:
+
+```python
+import onnx  
+import onnxruntime 
+
+print("ONNX version:", onnx.version)  
+print("ONNX Runtime version:", onnxruntime.__version__)  
+```
+Now, run version.py 
+
+```console
+$ python3 version.py
+```
+output:
+```output
+ONNX version: 1.18.0
+ONNX Runtime version: 1.22.0
+```
+### Download and Validate an ONNX Model :
+SqueezeNet is a lightweight convolutional neural network (CNN) architecture designed to achieve comparable accuracy to AlexNet, but with fewer parameters and smaller model size. 
+
+```console
+$ wget https://github.com/onnx/models/raw/main/validated/vision/classification/squeezenet/model/squeezenet1.0-12-int8.onnx -O squeezenet-int8.onnx
+```
+Validate the model: 
+
+Create a **vaildation.py** file with the code below for validation for ONNX model. 
+
+```python
+import onnx
+
+model = onnx.load("squeezenet-int8.onnx")
+onnx.checker.check_model(model)
+print("✅ Model is valid!")
+```
+output:
+```output
+✅ Model is valid!
+```
+This downloads a quantized (INT8) classification model on the both VMs, and validates its structure using ONNX’s built-in checker. 
+
+### Baseline testing using ONNX Runtime: 
```

Comment:
Explain in 1-2 lines what this test is, and what parameters it calculates to justify our testing.